e2e_test/source/basic/kafka.slt

```diff
@@ -326,6 +326,15 @@ create source s23 (bytea_example bytea) with (
   scan.startup.mode = 'earliest'
 ) row format json
 
+statement ok
```

Comment:
add some `statement error` cases, such as `create source s24 (id bytea, other field int)`
rw should reject the query.